subsys/net/ip/rpl.c

```diff
@@ -650,12 +650,20 @@ static void dio_timer(struct k_work *work)
 				net_if_ipv6_get_ll(rpl_default_iface,
 						   NET_ADDR_PREFERRED);
 
-			net_rpl_dio_send(rpl_default_iface, instance, addr,
-					 NULL);
+			if (addr) {
+				net_rpl_dio_send(rpl_default_iface, instance,
+						 addr, NULL);
+				#if defined(CONFIG_NET_STATISTICS_RPL)
+				instance->dio_send_pkt++;
+				#endif
+			} else {
```

Comment:
The case (addr == NULL) will only occur if, for example, an interface(802.15.4) do not exist or something went wrong when initializing the driver. I do not know how much sense it makes to resubmit here. See also the else branch below (L667). @rveerama1 Can you comment on that?

Reply:
I agree with @jfischer-phytec-iot. This doesn't make sense to resubmit after timeout. You end up with same addr is NULL. It's better to through NET_ERR message and find the root cause of addr being NULL. Do you have any test case or sample app with this scenario other than coverity issue?

Reply:
I observed it when I did not plug in FRDM-CR20A properly while testing rpl_border_router. To provoke it, it is enough to try rpl_border_router or rpl_node on FRDM-K64F without a FRDM-CR20A.